app/precompiled.py

```diff
@@ -768,21 +761,20 @@ def _get_pages_with_notify_tag(src_pdf_bytes):
 
 
 def redact_precompiled_letter_address_block(pdf, address_regex):
-    options = pdf_redactor.RedactorOptions()
-
-    options.content_filters = []
-    options.content_filters.append((
-        re.compile(address_regex),
-        lambda m: " "
+    pdf.seek(0)  # make sure we're at the beginning
+    doc = fitz.open("pdf", pdf)
+    first_page = doc[0]
+
+    first_page.add_redact_annot(fitz.Rect(
```

Comment:
I'd like to leave that to follow-up work. The address replacement is more complex as we need to specify it's position, which is at odds with the bounding rectangle we're using here for redaction.

While it would be nice to do it in one go, I don't think we gain much by doing so:

- The related body of code is quite small, so any reduction in complexity is minimal.
- We use the related dependencies - Canvas, PdfFileReader - elsewhere (still needed).